rust/chains/tw_pactus/src/transaction.rs

```diff
@@ -0,0 +1,410 @@
+        let signature = private_key.sign(sign_bytes)?;
+
+        self.set_signatory(private_key.public(), signature.clone());
+
+        Ok(signature)
+    }
+
+    pub fn set_signatory(&mut self, public_key: PublicKey, signature: Signature) {
+        // Unset "Not Signed" flag
+        self.flags &= !FLAG_NOT_SIGNED;
+
+        self.public_key = Some(public_key);
+        self.signature = Some(signature);
+    }
+
+    pub fn id(&self) -> Vec<u8> {
+        blake2_b(&self.sign_bytes().unwrap_or_default(), 32).unwrap_or_default()
+    }
+
+    pub fn to_bytes(&self) -> SigningResult<Vec<u8>> {
+        let mut w = Vec::with_capacity(self.encoded_size());
+
+        self.encode(&mut w)?;
+
+        Ok(w.to_vec())
+    }
+
+    pub fn sign_bytes(&self) -> SigningResult<Vec<u8>> {
+        let mut w = Vec::new();
+        self.encode_with_no_signatory(&mut w)?;
+        let mut sign_bytes = w.to_vec();
+        sign_bytes.remove(0); // Remove flags
+
+        Ok(sign_bytes)
+    }
+
+    fn encode_with_no_signatory(&self, w: &mut dyn std::io::Write) -> Result<(), EncoderError> {
+        self.flags.encode(w)?;
+        self.version.encode(w)?;
+        self.lock_time.encode(w)?;
+        self.fee.encode(w)?;
+        self.memo.encode(w)?;
+        self.payload.payload_type().encode(w)?;
+        self.payload.encode(w)?;
+
+        Ok(())
+    }
+
+    fn is_signed(&self) -> bool {
+        self.flags & FLAG_NOT_SIGNED == 0
+    }
+}
+
+impl Encodable for Transaction {
+    fn encode(&self, w: &mut dyn std::io::Write) -> Result<(), EncoderError> {
+        self.encode_with_no_signatory(w)?;
+
+        if let Some(sig) = &self.signature {
+            sig.encode(w)?;
+        }
+
+        if let Some(pub_key) = &self.public_key {
+            pub_key.encode(w)?;
+        }
+
+        Ok(())
+    }
+
+    fn encoded_size(&self) -> usize {
+        let mut len = self.flags.encoded_size()
+            + self.version.encoded_size()
+            + self.lock_time.encoded_size()
+            + self.payload.payload_type().encoded_size()
+            + self.fee.encoded_size()
+            + self.memo.encoded_size()
+            + self.payload.encoded_size();
+
+        if let Some(sig) = &self.signature {
+            len += sig.encoded_size();
+        }
+
+        if let Some(pub_key) = &self.public_key {
+            len += pub_key.encoded_size();
+        }
+
+        len
+    }
+}
+
+impl Decodable for Transaction {
+    fn decode(r: &mut dyn std::io::Read) -> Result<Self, EncoderError> {
+        let flags = u8::decode(r)?;
+        let version = u8::decode(r)?;
+        let lock_time = u32::decode(r)?;
+        let fee = Amount::decode(r)?;
+        let memo = String::decode(r)?;
+        let payload_type = PayloadType::decode(r)?;
+        let payload = match payload_type {
+            PayloadType::Transfer => Box::new(TransferPayload::decode(r)?),
+            _ => return Err(EncoderError::ParseFailed("Unsupported payload")),
```

Comment:
Can we remove other `PyalodType` enum variants?

Reply:
To maintain compatibility with the core code, I suggest keeping it, as it can later be compared with the source code.